algorand.cpp

```diff
@@ -175,8 +175,26 @@ bool is_present(Address a) {
 bool is_present(LogicSig lsig) {
   return is_present(lsig.logic);
 };
+bool is_present(Subsig subsig) {
+  return is_present(subsig.public_key);
+};
 bool is_present(MultiSig msig) {
-  return msig.threshold > 0;
+  if (msig.threshold == 0) {
+    return false;
+  }
+  else {
+    int thresholdSigs = msig.threshold;
+
+    for (const auto& sig : msig.sigs) {
+      if (false == is_present(sig)) {
+        break;
+      }
+      else if ( true == is_present(sig.signature) ) {
+        --thresholdSigs;
+      }
+    }
+    return (0 == thresholdSigs) && is_present(msig.address());
```

Comment:
I don't think this is quite right.  If you have an msig of 4 addresses, with a threshold of 2, it's OK to have 3 signatures.  But this code will decrement thresholdSigs to -1 and the final check will fail.

I would be perfectly ok with is_present(MultiSig) being much "dumber". is_present is just trying to decide whether something exists, and should therefore be msgpacked. So I think:
```
bool is_present(MultiSig msig) {
  return msigs.size() > 0;
}
```

By way of comparison - the other is_present functions for signatures don't check correctness of their signatures in any way - they just figure out if they are present at all.

Reply:
@jannotti, yep not sure if you saw my previous comment. I originally had this but I think I misinterpreted your previous PR comment which led me to reimplement it like this. 

I absolutely agree with this simpler approach and will roll it back to the way it was!